Sonar integration

rgildein opened the pull request:
- update newer templates
- add sonar

Commit:
add new templates, but without pylint and mypy

Commit:
add sonar

Commit:
apply new linting configs

agileshaw commented on Makefile:
```diff
@@ -72,12 +76,12 @@ unittests:
-functional:
-	@echo "Executing functional tests(charm will be build during functional tests)"
-	@tox -e func -- ${FUNC_ARGS}
+functional: 
```
It seems that a `build` target is needed here.
```suggestion
functional: build
```

rgildein replied:
No it's not. The pytest-operator will handle charm build for us.

agileshaw replied:
Noted. Then could you please remove the trailing blank space at the end?
```suggestion
functional:
```

Pjack reviewed (approved):
LGTM

rgildein committed:
Remove empty spaces from Makefile

This was addressed also in bootstack-charms-spec
https://code.launchpad.net/~rgildein/bootstack-charms-spec/+git/bootstack-charms-spec/+merge/449206

Co-authored-by: TQ X <agileshaw@users.noreply.github.com>
